sdk/boms/spring-cloud-azure-dependencies/pom.xml

```diff
@@ -55,6 +55,12 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <!-- TODO (zhihaoguo): specify the version of azure-identity, we need to remove this before merging into main -->
```

Comment:
Pin the version of azure-identity to 1.6.0-beta.1 for we want to use the caching ability for MI.